src/methods/extract.jl

```diff
@@ -285,27 +277,8 @@ Base.@assume_effects :total function _maybe_add_fields(::Type{T}, props::NamedTu
         :index in K ? merge((; geometry=point, index=I), props) : merge((; geometry=point), props)
     else
         :index in K ? merge((; index=I), props) : props
-    end
-end
-
-function _extractrowtype(x, g; geometry, index, skipmissing, names, kw...)
-    I = if istrue(skipmissing)
-        Tuple{Int, Int}
-    else
-        Union{Missing, Tuple{Int, Int}}
-    end
-    _extractrowtype(x, g, I; geometry, index, skipmissing, names, kw...)
-end
-function _extractrowtype(x, g, ::Type{I}; geometry, index, skipmissing, names, kw...) where I
-    G = if istrue(skipmissing)
-        nonmissingtype(typeof(g))
-    else
-        typeof(g)
-    end
-    _extractrowtype(x, G, I; geometry, index, skipmissing, names, kw...)
+    end |> T
```

Comment:
Apparently this `|> T` was needed to avoid allocations when geometry is a namedtuple.

Reply:
Coyld try ::T instead of piping too, it's a bit clearer

Reply:
It already had a ::T, but it was still allocating in cases like https://github.com/rafaqz/Rasters.jl/pull/790#issuecomment-2408503919 and also https://github.com/rafaqz/Rasters.jl/pull/791#issuecomment-2411420187, though I'm not sure why. I think it's something with the compiler giving up on the nested namedtuple structures?

Reply:
Yeah I thought those were functionally the same thing. ::T should call `convert`, but maybe the constructor is better.

The stability is the compiler knowing anything unstable is also an error, maybe that's just clearer with |> for some reaon